src/providers/authProvider/index.ts

```diff
@@ -4,24 +4,22 @@ import { AuditType, trackEvent } from '../../utils/audit'
 import bcrypt from 'bcryptjs'
 import { decryptData, encryptData, generateSalt } from '../../utils/ecnryption'
 export const getToken = (): string | null => {
-  const JsonUser = localStorage.getItem(constants.TOKEN_KEY)
-  if (JsonUser !== null) {
-    const user = JSON.parse(JsonUser) as User & { salt: string }
-    const decryptedData = decryptData(`${user.id}${user.salt}`)
-    user.id = decryptedData.substring(
-      0,
-      decryptedData.length - user.salt.length
-    )
-    return JSON.stringify(user)
+  const encryptedUser = localStorage.getItem(constants.TOKEN_KEY)
+  const salt = localStorage.getItem(constants.SALT)
+  if (encryptedUser !== null && salt !== null) {
+    const decryptedData = decryptData(`${encryptedUser}`)
+    return decryptedData.substring(0, decryptedData.length - 32)
   } else return null
 }
 
-const setToken = (token: string): void => {
+const setToken = (token: string, salt: string): void => {
   localStorage.setItem(constants.TOKEN_KEY, token)
+  localStorage.setItem(constants.SALT, salt)
```

Comment:
But @TahaKhanAbdalli - doesn't thevalue of `salt` come from the environment variable?  Why are we storing it in the cookie?

Reply:
It is general practice to store salt along with the encrypted data. What we are getting from env is the key.

Reply:
Sure, ok @TahaKhanAbdalli